addons/web/static/src/core/model.js

```diff
@@ -0,0 +1,34 @@
+/** @odoo-module **/
+
+import { useBus } from "@web/core/bus_hook";
+
+const { core, hooks } = owl;
+const { EventBus } = core;
+const { useComponent } = hooks;
+
+export class Model extends EventBus {
+    /**
+     * @param {Object} env
+     */
+    constructor(env) {
+        super();
+        this.env = env;
+        this.setup();
+    }
+
+    setup() {}
+}
+
+/**
+ * @param {Object} params
+ * @param {any} params.Model @todo specify type
+ * @param {Function} [params.onUpdate]
+ * @returns {any} @todo specify type
+ */
+export function useModel(params) {
```

Comment:
Model is mandatory, we can extract it from `params` so the API could be
```js
/**
 * @template T
 * @param {new (env: any) => T} Model
 * @param {Object} [params]
 * @param {Function} [params.onUpdate]
 * @returns {T}
 */
export function useModel(Model, params = {}) {
    const component = useComponent();
    const model = new Model(component.env);
    useBus(model, "update", params.onUpdate || component.render);
    return model;
}
```

Reply:
good like this?